specs/CustomDownload.md

```diff
@@ -0,0 +1,775 @@
+            CHECK_FAILURE(args->GetDeferral(&deferral));
+
+            // This function can be called to show the download dialog and
+            // complete the event at a later time, allowing the developer to
+            // perform async work before the event completes.
+            m_completeDeferredDownloadEvent = [showDialog, deferral] {
+                showDialog();
+                CHECK_FAILURE(deferral->Complete());
+            };
+
+            return S_OK;
+        })
+    .Get(),
+&m_downloadStartingToken));
+
+// Update download progress
+// Register a handler for the `DownloadProgressSizeInBytesChanged` event.
+CHECK_FAILURE(download->add_DownloadProgressSizeInBytesChanged(
+    Callback<ICoreWebView2DownloadProgressSizeInBytesChangedEventHandler>(
+        [this](ICoreWebView2Download* download, IUnknown* args) -> HRESULT {
+            // Here developer can update UI to show progress of a download using
+            // `download->get_DownloadProgressSizeInBytes` and `download->get_DownloadSizeInBytes`.
+            return S_OK;
+        })
+    .Get(),
+&m_downloadProgressSizeInBytesChangedToken));
+
+// Register a handler for the `DownloadStateChanged` event.
+CHECK_FAILURE(download->add_DownloadStateChanged(
+    Callback<ICoreWebView2DownloadStateChangedEventHandler>(
+        [this](ICoreWebView2Download* download,
+        ICoreWebView2DownloadStateChangedEventArgs* args) -> HRESULT {
+            BOOL canResume = FALSE;
+            COREWEBVIEW2_DOWNLOAD_STATE downloadState;
+            CHECK_FAILURE(download->get_State(&downloadState));
+            switch (downloadState)
+            {
+            case DOWNLOAD_IN_PROGRESS:
+                break;
+            case DOWNLOAD_INTERRUPTED:
+                // Here developer can take different actions based on `args->InterruptReason`.
+                MessageBox(nullptr, L"Connection with file host was broken.", nullptr, MB_OK);
+                CHECK_FAILURE(download->get_CanResume(&canResume));
+                if (canResume)
+                {
+                    download->Resume();
+                }
+                else
+                {
+                    CompleteDownload(download);
+                }
+                break;
+            case DOWNLOAD_COMPLETE:
+                CompleteDownload(download);
+                break;
+            }
+            return S_OK;
+        })
+    .Get(),
+&m_downloadStateChangedToken));
+```
+## .NET/ WinRT
+```c#
+webView.CoreWebView2.DownloadStarting += delegate (object sender, CoreWebView2DownloadStartingEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        using (deferral)
+        {
+            if (_coreWebView2Settings == null)
+            {
+                _coreWebView2Settings = webView.CoreWebView2.Settings;
+            }
+            _coreWebView2Settings.ShouldDisplayDefaultDownloadDialog = false;
```

Comment:
Same note about doing this in the event handler vs once outside the event handler